### PR TITLE
Code snippet evaluation

### PR DESCRIPTION
**Pull request by Deepankar-promact**
**Fixed Issues**

#23099

**Files Changed**

TestConductController.cs - Added controller for evaluating code snippet
TestAnswerAC.cs - Change code type from string to Code object
Code.cs - Code model to store code source, inputs and language
Result.cs - Result model to store result from code base server
ITestConductRepository.cs - Added Interface Functions
TestConductRepository.cs- Added implementation of interface function

**Checks**

- [x] Naming Conventions
- [x] Server side code comments (proper English, grammar and no spelling mistakes)
- [x] Client side code comments (proper English, grammar and no spelling mistakes)
- [x] Unused variables, methods, blank spaces and name spaces. 
- [x] Optimal Code and code formatting
- [x] Commit History is proper, no merge is done. 
- [x] Commit/pull request messages should be proper to justify your feature
- [x] All test cases are executed provided by tester.

**Known Issues**

- Not known

**Comments**

NA

---
This change is [Reviewable](https://reviewable.io/reviews/promact/trappist/388)

### CODE REVIEW

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 1]
```diff
@@ -233,11 +238,85 @@ public TestConductRepository(TrappistDbContext dbContext, ITestsRepository testR
...
+
+            return result;
+        }
+
+        /// <summary>
+        /// Assess Code Snippet Question and update the marks in report table
+        /// </summary>
+        /// <param name="attendeeId">Test Attendee Id</param>
+        /// <param name="testAnswer">TestAnswer object</param>
+        private async Task AssessCodeSnippet(int attendeeId, TestAnswerAC testAnswer)
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: MAJOR Remove the unused private method 'AssessCodeSnippet'. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS1144)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs [hunk 2]
```diff
@@ -18,9 +20,11 @@ namespace Promact.Trappist.Repository.TestConduct
     public class TestConductRepository : ITestConductRepository
     {
         #region Private Variables
+        private const string CodeBaseServer = "http://localhost:56725/api/execute";
```
Comment (rushi216) [inline Trappist/src/Promact.Trappist.Repository/TestConduct/TestConductRepository.cs]: we'll use code server url from appsettings value

File: Trappist/src/Promact.Trappist.Web/Migrations/20170805102902_Added_TestCodeSolution.cs [hunk 3]
```diff
@@ -0,0 +1,60 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore.Migrations;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace Promact.Trappist.Web.Migrations
+{
+    public partial class Added_TestCodeSolution : Migration
```
Comment (chintans) [inline Trappist/src/Promact.Trappist.Web/Migrations/20170805102902_Added_TestCodeSolution.cs]: MINOR Rename class 'Added_TestCodeSolution' to match camel case naming rules, consider using 'AddedTestCodeSolution'. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS101)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: Trappist/src/Promact.Trappist.Web/Migrations/20170805102902_Added_TestCodeSolution.cs (hunk 3 above)
Comment (chintans) [inline Trappist/src/Promact.Trappist.Web/Migrations/20170805102902_Added_TestCodeSolution.cs]: MINOR Rename class 'Added_TestCodeSolution' to match camel case naming rules, consider using 'AddedTestCodeSolution'. [rule](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS101)

**Comment by chintans** [comment]
SonarQube analysis reported 1 issue
* ![MINOR][MINOR] 1 minor

Watch the comments in this conversation to review them.

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**